python-scripts/xbox/interface/if_xbdm.py

```diff
@@ -77,21 +77,30 @@ def xbdm_command(cmd, length=None):
   return lines
 
 def GetModules():
-  return [] # FIXME: Make this work..
+  modulesList = {}
   lines = xbdm_command("modules")
-  for line in lines:
-    print(line)
-    chunks = line.split()
-    for chunk in chunks:
-      if (chunk[0:6] == b'name=\"'):
-        print("  name = '" + str(chunk[6:-1]) + "'")
-      elif (chunk[0:5] == b'base='):
-        print("  base = '" + str(chunk[5]) + "'")
-      else:
-        print('  unparsed: ' + str(chunk))
-        pass
-        #assert(False)
-  return []
+  lines = lines.decode("utf-8") 
+
+  linesList = lines.splitlines() # if we get multiline response -> #name="xboxkrnl.exe" #name="xbdm.dll" etc
+
+  for line in linesList: 
+
+    nameAddrDict ={}
+
+    nameAddrChunks = line.split()
+
+    for nameAddrChunk in nameAddrChunks:
+       singleChunk = nameAddrChunk.split("=")    #name="xbdm.dll" ->  ['name', '"xbdm.dll"']
+
+       if(singleChunk[0] =="name"):   #the name is key
+         modulesList[singleChunk[1].strip("\"")]=nameAddrDict #name of the module is the key(with out ' " ')
+                                                              #and its value is the dict with "Data" 
+                                                              #e.g. ["base"] = 0x80010000
```

Comment:
The quote symbols show a string which can contain spaces (presumably). There should be a comment about this being incomplete